### DISCUSSION TITLE
Markdown component editor language?

### DISCUSSION THREAD

**Discussion by zagoskin** [linkdotnet/Blog#383] [Q&A] [answered]
Hi! First of all I want to thank you for your effort. I'm loving the solution (i hate wordpress too). I'm passionate about C# and .NET in general and seeing it put to practice to this extent just makes me super happy and excited.

I was playing around with the blog and I noticed the text in the "Create blog" (or edit) markdown editors is being highlighted in red. I think this is due to Blazorise using CodeMirror internally and thinking I'm making a mistake, because the text is in spanish.

I've also confirmed that typing in english does not highlight my text.

I'm not very familiar with Blazorise or CodeMirror in general so I was seeking help as to how I can configure this so that it doesn't highlight text in spanish.

Things that I've tried but didn't work

- Setting the lang attribute for the _layout html
- Changing the CultureInfo properties to the spanish culture
- Using blazorise ITextLocalizer to set the language to spanish
- I'm out of ideas

Thanks for anyone that is able to provide help!

P.S: if you anyone also knows how to edit colors in general of the markdown editor that'd be great but this would just be a great plus. I don't mind it too much.

**Comment by linkdotnet** [comment, marked as answer]
Hey @zagoskin 

Thanks for your kind words. There are some measures we/you could do (thinking of adding some of those to the mainline branch). If we have a look at https://blazorise.com/docs/extensions/markdown (which is the markdown editor I am using). There are some tweaks:

 * `NativeSpellcheck`: If set to false, disable native spell checker. Defaults to true.
 * `SpellChecker`: If set to false, disable the spell checker. Defaults to true

Setting those to `false` will shut down the whole thing.

> if you anyone also knows how to edit colors in general of the markdown editor

Can you elaborate on this one?

EDIT:
To disable the whole spellcheck, edit the [`MarkdownTextarea Component`](https://github.com/linkdotnet/Blog/blob/master/src/LinkDotNet.Blog.Web/Features/Components/MarkdownTextArea.razor):
```html
<Markdown ElementId="@Id"
          Placeholder="@Placeholder"
          Class="@Class"
          Value="@Value"
          ValueChanged="s => Value = s"
          MaxHeight="@Height"
          PreviewRender="PreviewFunction"
          ImageAccept="image/*"
          ImageUploadChanged="UploadFiles"

          NativeSpellcheck="false"
          SpellChecker="false"

          PreviewImagesInEditor="true"/>
```

**Reply by zagoskin** [reply]
Thanks a bunch! I applied the code you suggested and it seems to work perfectly. A shame, though, that I couldn't get it to work but just for spanish. But still I wasn't even expecting for it to have a spellcheck so it's all good.

About the colors, what I meant is that the Markdown component has a toolbar for all the markdown actions, like setting text to bold, italic, heading, etc. This toolbar seems to have colors based on the theme (although for some reason mine is always gray, but in the Blazorise site it's either white or black depending on theme).

I was just wondering if I could easily edit some of the css for the toolbar. I looked at https://blazorise.com/docs/extensions/markdown and saw that it can be kinda modified to specify custom actions. But can't find where I could edit the css. I tried just adding the toolbar and setting the class/style attributes but apparently I can't.

I'll keep looking into it!

**Reply by linkdotnet** [reply]
To answer your question, have a look at [basic.css](https://github.com/linkdotnet/Blog/blob/master/src/LinkDotNet.Blog.Web/wwwroot/css/basic.css#L143):
```css
/* Fixes white background/white icon for markdown editors */
.editor-toolbar {
	background: rgb(36, 90, 170) !important;
}

.editor-toolbar button.active, .editor-toolbar button:hover {
    background: rgb(36, 90, 170) !important;
}
```

You can update the colors there. I just updated the blog (so on your fork, try to merge my latest `master`):
<img width="476" alt="image" src="https://github.com/user-attachments/assets/8256e600-a9bd-44ec-9dfe-bc9f8e776dc1">

**Reply by linkdotnet** [reply]
I am not sure if setting just the `NativeSpellchecker` to `true` Might pick something up that resolves to "Spanish dictionary" for you.

**Reply by zagoskin** [reply]
Oh that's just wonderful! My hero!

I'm honestly having a blast with your solution. Everything's been easy to understand and also I like the degree of customization that we have. Already telling all my friends that do c# about it.

**Reply by linkdotnet** [reply]
Really really appreciate your kind words! Thanks so much. Let me know if you need anything. Always a treat!

**Reply by linkdotnet** [reply]
Maybe more specifics why on the blazorise website the toggle is working but not here: the basic reason is that I am using bootstrap 5 and the bootstrap data attributes - and blazorise uses a different mechanism.

not tested, but something like this should work:
```css
html[data-bs-theme='light'] .editor-toolbar {…}
```

For the light theme settings and ‚dark‘, well, for the dark mode. 

If you want to customize the colors

**Reply by zagoskin** [reply]
That's great! Gonna try with that one. 

I also added tailwindcss, but I installed it by hand instead of doing it through blazorise as I was a bit scared it could break some of the stuff going. So i'll hook the theme selector to that attribute too. 

Awesome!

**Reply by linkdotnet** [reply]
That was one of the things I was thinking of as well - migrating to tailwindcss. But that takes just an immense amount of time.

**Reply by zagoskin** [reply]
Yeah don't worry about it. Your setup is fine and I was able to add tailwind with a prefix and have both of them working side by side so whatever custom stuff I wanna add I'm free to use either bs or tailwind.

The good thing is that your solution is not limiting me to use bootstrap/vanilla css.
